public/app/core/services/backend_srv.ts

```diff
@@ -296,8 +296,16 @@ export class BackendSrv implements BackendService {
 
   showSuccessAlert<T>(response: FetchResponse<T>) {
     const { config } = response;
+    const data: { message: string } = response.data as any;
 
+    // for data source configs we want to handle the app errors separately so emit a different event
+    // that will be handled outside of current notifications
     if (config.showSuccessAlert === false) {
+      // We don't need to show this to the user as the test status will be updated anyway and
+      // this causes flickering of the alert box when states are changing rapidly
+      if (data?.message && data.message !== 'Datasource updated') {
+        this.dependencies.appEvents.emit(DataSourceConfigEvents.success, ['success', data.message]);
+      }
```

Comment:
Does this mean that if any call-site suppresses success alerts (`config.showSuccessAlert = false`) and the data object has a `message` property other than `"Datasource updated"`, then it is going to emit a `DataSourceConfigEvents.success` event?

Reply:
the biggest question is why is there a DataSourceConfigEvents.success event, and why is emitted from BackendSrv?

Reply:
I just copied what we do for other errors/success by emmiting an event, but since it's a legacy thing I'll just do it with try/catch.